vue-todo-list/src/components/TodoDisplay.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <table>
+    <thead>
+      <tr>
+        <th class="todo-it">ID</th><th class="comment-col">コメント</th><th>状態</th><th class="no-bg"></th>
+      </tr>
+    </thead>
+    <tbody>
+        <tr v-for="(todo, index) in todos" :key="index">
+          <template v-if="showState === 'all' || showState === todo.state">
+            <td>{{ index }}</td>
+            <td>{{ todo.comment }}</td>
+            <td v-if="todo.state === 'wip'">
+              <button @click="changeState(index)">作業中</button>
+            </td>
+            <td v-else>
+              <button @click="changeState(index)">完了</button>
+            </td>
+            <td><button @click="deleteTodo(index)">削除</button></td>
+          </template>
+        </tr>
+    </tbody>
+  </table>
+</template>
+
+<script>
+export default {
+  props: ['todos', 'showState'],
+  methods: {
```

Comment:
以下の、deleteTodoメソッドと、changeStateメソッドは親コンポーネントで定義しているプロパティを変更するものなので、$emitで親側で削除や、変更をするものと認識していたのですが、ここで処理ができてしまいました。

TodoInputでTodoの登録や、TodoFilterで表示するTodoの状態の選択を変更するときは下記のエラーが出ていたのですが、ここでは出ないのはなぜでしょうか？？
TodoInputやTodoFilterでは、最初v-modelでpropsを変更させようとして下記エラーがでていたので、そこが怪しいと思っているのですが、確信にたどり着けていません。
`[Vue warn]: Avoid mutating a prop directly since the value will be overwritten whenever the parent component re-renders. Instead, use a data or computed property based on the prop's value. Prop being mutated:`
...

Reply:
warnningは出ていませんが、こちらのコンポーネントでは、おっしゃる通り、todosの表示のみで、詳細の処理などは親要素で実装した方が良いですね。